pkg/controllers/raycluster_webhook_test_support.go

```diff
@@ -0,0 +1,43 @@
+/*
+Copyright 2023.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import rayv1 "github.com/ray-project/kuberay/ray-operator/apis/ray/v1"
+
+func mutateWorkerGroupSpecs(rayCluster *rayv1.RayCluster) {
```

Comment:
Why did you move this function into separate file?

Reply:
The changes in this PR affect parts of the webhook [Default function](https://github.com/project-codeflare/codeflare-operator/blob/main/pkg/controllers/raycluster_webhook.go#L68-L124). I thought for testing that specific block of code, I would need to separate it and call that function on its own. The other approach could be to leave it where it was and assert an exact output or assert the length of items perhaps.

Reply:
Thinking whether it would have sense to verify result of the "main" function instead of checking every utility function....
i.e. call `Default(...)` or `ValidateCreate(...)` and so on.

Reply:
That approach will make tests independent on internal implementation functions in webhook, also it will verify whole workflow for the test cases.

Reply:
I think I get what you mean. So instead of calling the utility functions in the tests, I should call the "main" functions which will call those utility functions too. This way, basically testing the expected output from the functionality of the whole file right? - Sounds good to me, thanks for advising!

Reply:
right